Build failed in :test in Ubuntu

Hi,
I was trying to build clutz in Ubuntu 14.04

This is the error log that I am getting
```
:compileJava UP-TO-DATE
:processResources NO-SOURCE
:classes UP-TO-DATE
:jar UP-TO-DATE
:clutz UP-TO-DATE
:gents UP-TO-DATE
:startScripts SKIPPED
:distTar UP-TO-DATE
:distZip UP-TO-DATE
:assemble UP-TO-DATE
:compileTestJava UP-TO-DATE
:processTestResources NO-SOURCE
:testClasses UP-TO-DATE
:test

com.google.javascript.gents.TypeScriptGeneratorMultiTests > module_imports FAILED
    org.junit.ComparisonFailure: expected:<...   "originalName": "[both.A.B.C.D",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_both.js",
        "defaultRename": "D"
      },
      {
        "originalName": "both.A.B.C.D.x",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_both.js",
        "defaultRename": "D"
      },
      {
        "originalName": "both.A.B.C.D.bar",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_both.js",
        "defaultRename": "D"
      },
      {
        "originalName": "default.A",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_default.js",
        "defaultRename": "A"
      },
      {
        "originalName": "named.A.B",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "named.A.B.x",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "named.A.B.foo",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "named.A.B.bar",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B.x",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B.foo",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B.bar",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "sideeffect.A.B.C",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_sideeffect.js",
        "defaultRename": ""
      },
      {
        "originalName": "no.empty.export.with.import",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/import_assignment.js",
        "defaultRename": ""
      },
      {
        "originalName": "import.binding.barrel",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/import_binding.js",
        "defaultRename": "]"
      }
    ]> but was:<...   "originalName": "[named.A.B",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "named.A.B.x",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "named.A.B.foo",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "named.A.B.bar",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_named.js",
        "defaultRename": ""
      },
      {
        "originalName": "no.empty.export.with.import",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/import_assignment.js",
        "defaultRename": ""
      },
      {
        "originalName": "sideeffect.A.B.C",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_sideeffect.js",
        "defaultRename": ""
      },
      {
        "originalName": "import.binding.barrel",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/import_binding.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B.x",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B.foo",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "namespace.A.B.bar",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_namespace.js",
        "defaultRename": ""
      },
      {
        "originalName": "both.A.B.C.D",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_both.js",
        "defaultRename": "D"
      },
      {
        "originalName": "both.A.B.C.D.x",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_both.js",
        "defaultRename": "D"
      },
      {
        "originalName": "both.A.B.C.D.bar",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_both.js",
        "defaultRename": "D"
      },
      {
        "originalName": "default.A",
        "tsFile": "src/test/java/com/google/javascript/gents/multiTests/module_imports/export_default.js",
        "defaultRename": "A]"
      }
    ]>
:test FAILED

BUILD FAILED

Total time: 20.473 secs
```

Any Ideas on how to fix it?

--- a/src/main/java/com/google/javascript/gents/ModuleRenameLogger.java
+++ b/src/main/java/com/google/javascript/gents/ModuleRenameLogger.java
@@ -5,6 +5,7 @@ import com.google.gson.GsonBuilder;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 
 /**
@@ -23,7 +24,7 @@ import java.util.Set;
  * <p>Which generates the following log line: foo.bar,buz.ts,A
  */
 class ModuleRenameLogger {
-  private class LogItem {
+  public class LogItem {
     String originalName;
     String tsFile;
     String defaultRename;
@@ -32,6 +33,31 @@ class ModuleRenameLogger {
       this.originalName = originalName;
       this.tsFile = tsFile;
       this.defaultRename = defaultRename;
+    }
+
+    public boolean equals(Object obj) {
+      if (!(obj instanceof LogItem)) return false;
+      LogItem o = (LogItem) obj;
+      return this.originalName.equals(o.originalName)
+          && this.tsFile.equals(o.tsFile)
+          && this.defaultRename.equals(o.defaultRename);
+    }
+
+    public String toString() {
+      return "{originalName: "
+          + this.originalName
+          + ",\n"
+          + "{tsFile: "
+          + this.tsFile
+          + ",\n"
+          + "{defaultRename: "
+          + this.defaultRename
+          + "}";
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(this.originalName, this.tsFile, this.defaultRename);
     }
   }
 

--- a/src/test/java/com/google/javascript/gents/TypeScriptGeneratorMultiTests.java
+++ b/src/test/java/com/google/javascript/gents/TypeScriptGeneratorMultiTests.java
@@ -3,9 +3,11 @@ package com.google.javascript.gents;
 import static com.google.common.truth.Truth.assertThat;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.io.Files;
 import com.google.gson.Gson;
 import com.google.javascript.clutz.DeclarationGeneratorTests;
+import com.google.javascript.gents.ModuleRenameLogger.LogItem;
 import com.google.javascript.gents.TypeScriptGenerator.GentsResult;
 import com.google.javascript.jscomp.SourceFile;
 import java.io.ByteArrayOutputStream;
@@ -108,9 +110,14 @@ public class TypeScriptGeneratorMultiTests extends TypeScriptGeneratorTests {
         File logFile = getTestDirPath(multiTestPath).resolve(dirName).resolve("log.json").toFile();
         if (logFile.exists()) {
           String goldenLog = getFileText(logFile);
-          // asserts that golden parses correctly as JSON, result is not needed.
-          new Gson().fromJson(goldenLog, Object.class);
-          assertThat(gentsResult.moduleRewriteLog).isEqualTo(goldenLog);
+          // order independent comparison of the log entries.
+          Set<LogItem> goldenLogItems =
+              ImmutableSet.copyOf(new Gson().fromJson(goldenLog, LogItem[].class));
+          Set<LogItem> emittedLogItems =
+              ImmutableSet.copyOf(
+                  new Gson().fromJson(gentsResult.moduleRewriteLog, LogItem[].class));
+
+          assertThat(emittedLogItems).isEqualTo(goldenLogItems);
         }
       } catch (Throwable t) {
         result.addError(this, t);

